airflow/bin/cli.py

```diff
@@ -85,7 +85,10 @@ def trigger_dag(args):
 
     session = settings.Session()
     # TODO: verify dag_id
-    execution_date = datetime.now()
+    if args.execution_date:
+        execution_date = dateutil.parser.parse(args.execution_date)
```

Comment:
Can you catch the exception if the execution date can not be parsed/provide a meaningful error message and exit?